pytest_inmanta/plugin.py

```diff
@@ -735,7 +735,10 @@ def get_handler(self, resource: Resource, run_as_root: bool) -> ResourceHandler:
 
     def finalize_context(self, ctx: handler.HandlerContext) -> None:
         # ensure logs can be serialized
-        json_encode({"message": ctx.logs})
+        try:
+            json_encode({"message": ctx.logs})
```

Comment:
Why do we need this change? Is it just to improve the error reporting to the user? In that case the changelog entry doesn't really describe the nature of the change correctly.

Reply:
The ticket mentions this:
>3. we should update pytest-inmanta to check in finilize context of invalid items are pressent.

This is how I understood it: raise an exception if somehow some unserializable items are still present in the logs. It's possible I misinterpreted it though.

Reply:
Oh I think I see. So since the check is already there I can close this ?

Reply:
I think so. But I am not entirely sure about the intention here. The fix on inmanta-core will also ensure that no exception is raised when a non-serializable item is used. I think it would be good to cross check this with Wouter.

Reply:
ticket #3470 is not about logs, but about the changes dict. 

If the changes in core make sure it can never be wrong, nothing is required here.